async-nats/src/jetstream/message.rs

```diff
@@ -172,6 +173,130 @@ impl Message {
             )))
         }
     }
+
+    /// Returns the `JetStream` message ID
+    /// if this is a `JetStream` message.
+    #[allow(clippy::eval_order_dependence)]
+    pub fn jetstream_message_info(&self) -> Result<JetStreamMessageInfo<'_>, Error> {
```

Comment:
Some bike shedding on this since it's pub, token_info or something like that?

```suggestion
    pub fn info(&self) -> Result<JetStreamMessageInfo<'_>, Error> {
```

Reply:
Fixing it means refactoring the macro IMO.

Let's put it on board to do that sometime soon.

Reply:
If we know we're gonna rename it, hide it for now.

Reply:
I renamed to `info` and that seems fine. I wanted to punt refactoring of macro only.